python/services/dataservice/dmod/dataservice/service_settings.py

```diff
@@ -1,20 +1,203 @@
-from pydantic import BaseSettings, Field
-from typing import Optional
-from typing_extensions import Annotated, TypeAlias
+from pathlib import Path
+from socket import gethostname
+from typing import Optional, Type
 
-Port: TypeAlias = Annotated[int, Field(..., ge=0, le=65535)]
+from pydantic import BaseSettings, ConstrainedInt, DirectoryPath, Field, FilePath
+
+
+class Port(ConstrainedInt):
+    ge = 0
+    le = 65535
 
 
 class ServiceSettings(BaseSettings):
     s3fs_url_protocol: str = "http"
     s3fs_url_host: Optional[str]
-    s3fs_url_port: Port = 9000
+    s3fs_url_port: Port = Field(9000)
     s3fs_vol_image_name: str = "127.0.0.1:5000/s3fs-volume-helper"
     s3fs_vol_image_tag: str = "latest"
     s3fs_plugin_alias: str = "s3fs"
     s3fs_helper_network: str = "host"
 
+    host: str = Field(default_factory=gethostname)
+    """Set the appropriate listening host name or address value (NOTE: must match SSL cert)"""
+    port: Port = Field(3012)
+    """Set the appropriate listening port value"""
+
+    cert_path: Optional[FilePath] = None
+    """Specify path for a particular SSL certificate file to use"""
+    key_path: Optional[FilePath] = None
+    """Specify path for a particular SSL private key file to use"""
+    ssl_dir: Optional[DirectoryPath] = None
+    """Change the base directory when using SSL certificate and key files with default names"""
+
+    object_store_host: str = "minio-proxy"
+    """Set hostname for connection to object store"""
+    object_store_port: Port = Port(9000)
+    """Set port for connection to object store"""
+
+    # TODO: consider making this a pydantic.SecretStr
+    object_store_username: str
+    """Set name of the Docker secret containing the object store user access key"""
```

Comment:
We probably shouldn't engineer something at this level in a way that is quite so tailored to our use of Docker.  I suggest we rename this and the other similar items to suggest that they are really the names of files (at least as far as the app is concerned), not only to be more clear, but so we can have separate attributes for the values.  We may not always acquire the value in this manner in the future.

Reply:
> We probably shouldn't engineer something at this level in a way that is quite so tailored to our use of Docker

Are you primarily suggesting that I should change the doc string to remove the reference to Docker? I did not make it clear in the source documentation that any defined field can be sourced programmatically (when initializing the object), from an environment variable, from a `.env` file, or from a file with the fields name stored in `/run/secrets` (following Docker's convention). I don't think we should do this now, but if we _need_ to configure _where_ secrets are sourced, there is a straight forward pathway to do that.

Reply:
The docstring was part of it.  My general concern was that if `object_store_username`/`object_store_exec_user_name` stored the name of a Docker secret, then using that attribute name would be misleading unless we assumed certain things about having Docker in the deployment.  

But from your response, it sounds like somehow  _ServiceSettings_  gets the content of secret files automatically as appropriate.  I'm guessing that it knows to puts those contents into `object_store_exec_user_name`, instead of the value directly provided, when the provided value is actually something in the secrets dir.  Alternatively, if the value within an `.env` file for `object_store_exec_user_name` is just a literal value for the user name, the _ServiceSettings_ object can tell the difference and just use the literal value.  I just don't quite see how that works.  

As long as that is how it works, then this is fine (apart from the elsewhere discussed `exec` stuff for the name).  And that would explain why the environment variables themselves aren't named to reflect that they are Docker secret names.   Though the way things all fit together is perhaps a bit obfuscated and could use some clarifications (certainly it was with the previous docstring, though I see you've already changed that).

Reply:
Yep, you nailed it. That is pretty much exactly how it works. I tried to emphasize that in the README. Im sure ive missed the mark somewhere. Can you skim it and let me know holes i've left? I want to make sure it is clear how to configure the service.